### DISCUSSION TITLE
is variable's type decide at runtime or compile time?

### DISCUSSION THREAD

**Discussion by ravinder387** [flix/flix#8485] [Q&A] [answered]
let x = 1; give error of parsing error
![Screenshot from 2024-07-31 06-29-35](https://github.com/user-attachments/assets/fcd8f7a7-baf2-4a4d-812c-0ab629e3cbc9)

**Comment by magnus-madsen** [comment, marked as answer]
Hi @ravinder387 and welcome to our community.

A variable's type is decided at compile-time.

Could you show your full program?
